include/dxc/DXIL/DxilInstructions.h

```diff
@@ -8774,28 +8773,17 @@ struct DxilInst_BarrierByMemoryType {
                              llvm::IntegerType::get(Instr->getContext(), 32),
                              llvm::APInt(32, (uint64_t)val)));
   }
-  llvm::Value *get_AccessFlags() const { return Instr->getOperand(2); }
-  void set_AccessFlags(llvm::Value *val) { Instr->setOperand(2, val); }
-  int32_t get_AccessFlags_val() const {
+  llvm::Value *get_SemanticFlags() const { return Instr->getOperand(2); }
+  void set_SemanticFlags(llvm::Value *val) { Instr->setOperand(2, val); }
```

Comment:
Wouldn't we want to use OperandIdx::arg_SemanticFlags here instead of "2" ?

Reply:
This is generated code, so the number should be correct.  The `OperandIdx::arg_SemanticFlags` are for use outside the generated code.  We could update the code generator to use these instead (and cast), but I don't think that would have any real benefit, and it certainly wouldn't be something to do in this PR.